### PR TITLE
Fix TransportDesiredNodesActionsIT batch tests

### LINKED ISSUE #83386: [CI] TransportDesiredNodesActionsIT testDeleteDesiredNodesTasksAreBatchedCorrectly failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/3sdeo4fn4dpio/tests/:server:internalClusterTest/org.elasticsearch.action.admin.cluster.desirednodes.TransportDesiredNodesActionsIT/testDeleteDesiredNodesTasksAreBatchedCorrectly

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.action.admin.cluster.desirednodes.TransportDesiredNodesActionsIT.testDeleteDesiredNodesTasksAreBatchedCorrectly" -Dtests.seed=EFDF9BFF1D476AA8 -Dtests.locale=ro-RO -Dtests.timezone=US/Indiana-Starke -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.action.admin.cluster.desirednodes.TransportDesiredNodesActionsIT&tests.test=testDeleteDesiredNodesTasksAreBatchedCorrectly

**Failure excerpt:**
```
java.lang.Exception: Test abandoned because suite timeout was reached.

  at __randomizedtesting.SeedInfo.seed([EFDF9BFF1D476AA8]:0)

```

### PR DESCRIPTION
**Pull request by fcofdez**
Select the current master to submit the blocking cluster state
update task, otherwise the task never gets executed.

Closes #83386

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/desirednodes/TransportDesiredNodesActionsIT.java [hunk 1]
```diff
@@ -372,7 +370,7 @@ private UpdateDesiredNodesResponse updateDesiredNodes(DesiredNodes desiredNodes)
     private Runnable blockClusterStateUpdateThread() throws InterruptedException {
         final CountDownLatch unblockClusterStateUpdateTask = new CountDownLatch(1);
         final CountDownLatch blockingClusterStateUpdateTaskExecuting = new CountDownLatch(1);
-        final ClusterService clusterService = internalCluster().getMasterNodeInstance(ClusterService.class);
+        final ClusterService clusterService = internalCluster().getCurrentMasterNodeInstance(ClusterService.class);
```
Comment (DaveCTurner) [inline server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/desirednodes/TransportDesiredNodesActionsIT.java:373]: This API is so trappy.

File: server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/desirednodes/TransportDesiredNodesActionsIT.java [hunk 2]
```diff
@@ -383,7 +381,8 @@ public ClusterState execute(ClusterState currentState) throws Exception {
 
             @Override
             public void onFailure(Exception e) {
-
+                blockingClusterStateUpdateTaskExecuting.countDown();
```
Comment (DaveCTurner) [inline server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/desirednodes/TransportDesiredNodesActionsIT.java:384]: Recommend also bounding the wait time (`assertTrue(blockingClusterStateUpdateTaskExecuting.await(10, TimeUnit.SECONDS));` and similarly for `unblockClusterStateUpdateTask`).

**Review by DaveCTurner — COMMENTED** [review]
Looks good, I left another suggestion too.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Review by henningandersen — APPROVED** [review]
LGTM2

**Pull request merged**